src/test/xmeta/test_symbol_table.cpp

```diff
@@ -0,0 +1,92 @@
+#include "pch.h"
+
+#include <algorithm>
+#include <iostream>
+#include <string_view>
+
+#include "antlr4-runtime.h"
+#include "symbol_table.h"
+#include "XlangParser.h"
+#include "XlangLexer.h"
+#include "ast_to_st_listener.h"
+#include "helpers.h"
+
+using namespace antlr4;
+using namespace xlang::xmeta;
+
+xmeta_symbol_table st{ "" };
+
+TEST_CASE("Duplicate Namespaces")
+{
+    std::string test_idl =
+        "namespace n { } \
+        namespace n { }";
+
+    st.reset("dup_ns.idl");
+    ast_to_st_listener listener{ st };
+    REQUIRE(setup_and_run_parser(test_idl, listener) == 0);
+
+    auto namespaces = st.get_namespaces();
+    auto it = namespaces.find("n");
+    REQUIRE(it != namespaces.end());
+    auto ns = it->second;
+    REQUIRE(ns->get_namespace_bodies().size() == 2);
+    REQUIRE(ns->get_namespace_bodies()[0]->get_containing_namespace() == ns);
+    REQUIRE(ns->get_namespace_bodies()[1]->get_containing_namespace() == ns);
+}
+
+TEST_CASE("Enum test")
+{
+    std::string test_idl =
```

Comment:
Nit: To avoid all those backslashes you can use a raw string literal:
```c++
std::string test_idl = R"abc(
namespace n {}
namespace n{}
)abc";
```